packages/color-scale-block/src/components/SquareWithColor.tsx

```diff
@@ -0,0 +1,117 @@
+/* (c) Copyright Frontify Ltd., all rights reserved. */
+
+import { useDrag } from 'react-dnd';
+import { DragSourceMonitor } from 'react-dnd';
+import { Button, ButtonSize, ButtonStyle, IconSize, IconTrashBin, Tooltip, useCopy } from '@frontify/fondue';
+import { ItemDragState, TooltipAlignment, TooltipPosition } from '@frontify/fondue';
+import { SquareWithColorProps } from '../types';
+import { DragHandle } from './DragHandle';
+import { TooltipContent } from './TooltipContent';
+import { joinClassNames, toHex8String } from '@frontify/guideline-blocks-shared';
+
+export const SquareWithColor = ({
+    id,
+    index,
+    width,
+    height,
+    isFirst,
+    isLast,
+    currentColor,
+    onResizeStart,
+    calculateLeftPosition,
+    isEditing,
+    deleteColor,
+    listId,
+    setCurrentlyDraggedColorId,
+    isDragging,
+}: SquareWithColorProps) => {
+    const handleDrag = () => {
+        if (isDragging !== !!currentColor.id) {
+            setCurrentlyDraggedColorId(currentColor.id ?? null);
+        }
+    };
+
+    const handleDragEnd = () => {
+        setCurrentlyDraggedColorId(null);
+    };
+
+    const [{}, drag] = useDrag({
+        item: currentColor,
+        collect: (monitor: DragSourceMonitor) => {
+            return {
+                componentDragState: monitor.isDragging() ? ItemDragState.Dragging : ItemDragState.Idle,
+            };
+        },
+        type: listId,
+        canDrag: isEditing,
+    });
+
+    const { copy, status } = useCopy();
+
+    const rgbaValues = [
+        currentColor.red,
+        currentColor.green,
+        currentColor.blue,
+        ...(currentColor.alpha ? [currentColor.alpha / 255] : []),
+    ];
+
+    const backgroundColor = `rgba(${rgbaValues.join(',')})`;
+    const hexColor = currentColor && toHex8String(currentColor);
```

Comment:
```suggestion
    const hexColor = currentColor ? toHex8String(currentColor) : '';
```
Keep the same type if possible